### ISSUE TITLE
Axes3D: factor out 3D coordinate guessing from format_coord()

### ISSUE THREAD

**Issue by ndawe** [matplotlib/matplotlib#4334]
To get the estimated (x,y,z) coordinate of a mouse event in a 3D figure we must parse these values from the string [generated](https://github.com/matplotlib/matplotlib/blob/e2315d221251e62130e3cd73c39fe90b5924600f/lib/mpl_toolkits/mplot3d/axes3d.py#L1087-1125) by ``Axes3D.format_coord``. Wouldn't it be more useful to factor that out into a separate method (`coord`?) giving access to `x, y, z = proj3d.inv_transform(xd, yd, z, self.M)`? `format_coord` would then simply convert that to the formatted string.

The question of how to extract these coordinates has been asked a few times on SO:

http://stackoverflow.com/questions/6748184/matplotlib-plot-surface-get-the-x-y-z-values-written-in-the-bottom-right-cor
http://stackoverflow.com/questions/28954611/how-to-extract-3d-mouse-coordinate-on-mouse-click-in-matplotlib-in-python

and aside from duplicating the code within `format_coord` the best we can do is parse the values out of the generated string.

**Comment by tacaswell** [comment]
attn @WeatherGod

**Comment by tacaswell** [comment]
I agree it would make sense to turn from Line 1101 to the end of the function into

```python
def guess_xzy(self, x, y):
    ....
```

and then use it within `format_coord`.

@ndawe Any desire to take a crack at implementing this?

**Comment by ndawe** [comment]
Sure, I'll open a PR.

**Comment by ndawe** [comment]
Haven't forgotten... I have time this weekend.
